pkg/virt-handler/virtwrap/manager.go

```diff
@@ -328,21 +327,14 @@ func VMNamespaceKeyFunc(vm *v1.VM) string {
 	return domName
 }
 
-func (l *LibvirtDomainManager) SyncVM(vm *v1.VM) (*api.DomainSpec, error) {
-	var wantedSpec api.DomainSpec
-	mappingErrs := model.Copy(&wantedSpec, vm.Spec.Domain)
-
-	if len(mappingErrs) > 0 {
-		return nil, errors.NewAggregate(mappingErrs)
-	}
-
+func (l *LibvirtDomainManager) SyncVM(vm *v1.VM, domDesign *designer.DomainDesign) (*api.DomainSpec, error) {
```

Comment:
It can't work that way - you can't separate the resolving of k8s objects like disks, from from the conversion to libvirt XML - that separation only works today because the public DomainSpec is identical to the internal DomainSpec (aka libvirt XML), so they are capable of holding the same data. This is not the case going forward, which is why the resolving os k8s objects is refactored in this series, so that we immediately create the internal DomainSpec (to be replaced by libvirt-go-xml object in later patches). The current code that deals with resolving Disk sources is already done in the VM handler, not SyncVM, and this refactoring keeps it at that point.